### PR TITLE
improvement(table): add rowBackground prop

### PR DESCRIPTION
**Pull request by swapnilmisal**
### JIRA Ticket

Jira:  [AGCT-1096](https://hello-haptik.atlassian.net/browse/AGCT-1096)

--

### Changelog

rowBackground prop enables alternate row background style

--

https://devswapnil.hellohaptik.com/relements/?path=/story/theme-spec--table-variants

### Screenshots (if appropriate):
image

### CODE REVIEW

**Review by gopal-virtual — APPROVED** [review]

File: src/components/Context/Theme.storybook.js [hunk 1]
```diff
@@ -138,6 +138,9 @@ storiesOf("Theme/Spec", module).add("Table variants", () => {
       <h3>Compact</h3>
       <Table themed space="compact" {...tableProps} />
       <br />
+      <h3>Sparse</h3>
+      <Table themed space="sparse" {...tableProps} />
```
Comment (gopal-virtual) [inline src/components/Context/Theme.storybook.js:142]: variant should match what designers have defined.

**Review by gopal-virtual — CHANGES_REQUESTED** [review]

File: src/components/Context/Theme.storybook.js (hunk 1 above)
Comment (swapnilmisal) [reply src/components/Context/Theme.storybook.js:142]: I don't see a problem here,
1. the quick resolutions could be that, I'll remove sparse and put this spacing in the default.
2. or I'll remove the sparse altogether and its spacing,-- but then some features will have to wait until the spacing is fixed

**Review by gopal-virtual — APPROVED** [review]

**Review by AwadheshJha — APPROVED** [review]

**Review by sidMulay — APPROVED** [review]

**Comment by swapnilmisal** [comment]
RETEST THIS PLEASE!

**Pull request merged**